src-ext/Typescript/ggvega/js/Encoding.js

```diff
@@ -13,11 +14,23 @@ export function TranslateEncoding(layer, labels, layerData, scales) {
     };
     return layerEncoding;
 }
-function TranslateXClass(layer, labels, layerData, scales) {
-    var field = layer['mapping']['x']['field'];
-    var type = layerData['metadata'][field]['type'];
+function TranslateXClass(layer, labels, layerData, scales, mark) {
+    var xClass;
     var scale;
     var title = labels['x'];
+    if (!layer['mapping']['x']) {
+        if (mark == Mark.Bar) {
+            xClass = {
+                aggregate: AggregateOp.Count,
```

Comment:
Just as a first impression, I don't think you need to hard-code `AggregateOp.Count`; this information is available in the `stat` element in the layer.

Reply:
Yes, you are right. I should use the information in `stat`.

However, I can not find all kinds of `stat` on the internet. Could you send me a link about all `stat` so that I can discover the relationship between `stat` and `aggregate` . So far I just consider `StatCount` and `StatSum`. 

I creat a new file `Stat.ts` to translate the `Stat`.